feat: add `stats/incr/nanmrss`

jalajk3004 opened the pull request:
Resolves #5607

## Description

> What is the purpose of this pull request?

This pull request:

-   {{TODO: add description describing what this pull request does}}

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5607

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

jalajk3004 committed:
edit

jalajk3004 committed:
edit

jalajk3004 committed:
Merge branch 'develop' of https://github.com/jalajk3004/stdlib into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
done

jalajk3004 committed:
lint

jalajk3004 committed:
added nanmrss

jalajk3004 committed:
added module

jalajk3004 committed:
added module

jalajk3004 committed:
lint

jalajk3004 committed:
lint

jalajk3004 committed:
lint

jalajk3004 committed:
lint

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmrss/README.md:
```diff
@@ -0,0 +1,186 @@
+
+#### incrnanmrss( window )
+
+Returns an accumulator `function` which incrementally computes a moving [residual sum of squares][residual-sum-of-squares]. The `window` parameter defines the number of values over which to compute the moving [residual sum of squares][residual-sum-of-squares].
```
Needs to add `, ignoring `NaN` values.` at the end.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmrss/docs/types/index.d.ts:
```diff
@@ -0,0 +1,83 @@
+/**
+* If provided arguments, returns an updated residual sum of squares; otherwise, returns the current residual sum of squares, ignoring `NaN` values.
+*
+* ## Notes
```
Remove from `L26-29`

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmrss/lib/main.js:
```diff
@@ -0,0 +1,92 @@
+	* @returns {(number|null)} residual sum of squares or null
+	*/
+	function accumulator( x, y ) {
+		if ( arguments.length === 0 ) {
```
Instead of using two if statements. This can be done in a single if statement. (L79-84)

hrshya reviewed (commented):
The `img` directory is also missing in `docs`.

jalajk3004 committed:
check final

jalajk3004 commented:
@hrshya you can review this PR again, i have resolved the errors

hrshya commented:
> @hrshya you can review this PR again, i have resolved the errors

On it!

hrshya reviewed (commented):
@jalajk3004 Can you also change the description to:

adds `stats/incr/nanmrss`  .

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmrss/README.md:
```diff
@@ -0,0 +1,186 @@
+
+# incrnanmrss
+
+> Compute a moving [residual sum of squares][residual-sum-of-squares] (RSS) incrementally, ignoring `NaN` calues.
```
Typo.

```suggestion
> Compute a moving [residual sum of squares][residual-sum-of-squares] (RSS) incrementally, ignoring `NaN` values.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmrss/lib/main.js:
```diff
@@ -0,0 +1,89 @@
+		if ( arguments.length === 0 || isnan( x ) || isnan( y ) ) {
+			return mrss();
+		}
+		return mrss( ( ( isnan( x ) ) ? 0 : x), ( ( isnan( y ) ) ? 0 : y ) );
```
Dead code. Line 79 already returns early when either `x` or `y` is NaN, so the `isnan` ternaries are unreachable.

```suggestion
		return mrss( x, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmrss/docs/types/index.d.ts:
```diff
@@ -0,0 +1,79 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrmrss( 3 );
```
Wrong function name.

```suggestion
* var accumulator = incrnanmrss( 3 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmrss/examples/index.js:
```diff
@@ -0,0 +1,41 @@
+	v1 = ( bernoulli( 0.8 ) < 1 ) ? NaN : uniform( 0.0, 100.0 );
+	v2 = ( bernoulli( 0.8 ) < 1 ) ? NaN : uniform( 0.0, 100.0 );
+	rss = accumulator( v1, v2 );
+	console.log( '%s\t%s\t%s', ((Number.isNaN(v1)) ? 'NaN' : v1.toFixed(3)), ((Number.isNaN(v2)) ? 'NaN' : v2.toFixed(3)), (((rss !== null) && (!Number.isNaN(rss))) ? rss.toFixed(3) : 'null'));
```
`Number.isNaN` is ES6. Examples must be ES5. Also missing spaces inside parentheses. See `mrss/examples/index.js`.

Planeshifter reviewed (changes requested):
Better late than never... thanks for opening a PR!

Planeshifter committed:
fix: address review feedback in stats/incr/nanmrss

- Remove unreachable NaN-guard ternaries in the accumulator return (the early guard already returns on NaN); forward x and y directly
- Fix incrmrss -> incrnanmrss in the index.d.ts example
- Use @stdlib/math/base/assert/is-nan instead of ES6 Number.isNaN in the example and add conventional spacing
- Fix the README 'calues' typo and correct misleading window annotations for ignored NaN pairs
- Drop backticks from package.json description; bump copyright year to 2026

Planeshifter reviewed (approved):
Looks good.

kgryte commented:
@Planeshifter Not sure why, but all the linting in this PR was a no-op in CI: https://github.com/stdlib-js/stdlib/actions/runs/28843282273/job/85541581494?pr=6197

Worthwhile to always merge in the latest `develop` before fixing up these older PRs.

kgryte commented:
I noticed this, as CI failed after merging to `develop`.
